### DISCUSSION TITLE
How to link frontmatter data in frontmatter

### DISCUSSION THREAD

**Discussion by Maarcel** [11ty/eleventy#2286] [Q&A]
I have a template `fundraising.njk` containing something like this (it is in german):

`<p>In dieser Spendenaktion haben bereits <b class="no-line-break">{{ fundraising.contributors}} Spender</b> mitgewirkt (Stand {{ page.date | formatDate({ month: '2-digit', day: 'numeric', hour: 'numeric', minute: 'numeric' }, true) }} Uhr). Wir sagen DANKE für euren Beitrag!</p>`

Now i have two files `de/fundraising.md` and `en/fundraising.md` that are using this layout.

What is the best practice to move this line of code from the template to the frontmatter markdown files so text can be translated?

I dont't know how to include `{{ fundraising.contributors}}` and `{{ page.date | formatDate({ month: '2-digit', day: 'numeric', hour: 'numeric', minute: 'numeric' }, true) }}` into the frontmatter.

Thanks in advance!

**Comment by pdehaan** [comment]
@Maarcel, not sure if I understand exactly what you're trying to do.
For something like this:
```njk
{{ page.date | formatDate({ month: '2-digit', day: 'numeric', hour: 'numeric', minute: 'numeric' }, true) }}
```

You could move the `formatDate()` config into the frontmatter using something like:

```njk
---
dateFmt:
	month: "2-digit"
    day: numeric
    hour: numeric
    minute: numeric
---

{{ page.date | formatDate(dateFmt, true) }}
```

Not convinced that's the ideal solution. Ideally you wouldn't be specifying that over and over in each template where you want to format dates. Although it would let you use custom date formats per-locale.

You could probably create some src/_data/dateformats.js global data file which just exports formats.
Maybe something like:

```js
module.exports = {
  en: {
    short: { month: "2-digit", day: "numeric", hour: "numeric", minute: "numeric" },
    long: { ... }
  },
  de: {
    short: { month: "...", ... },
    long: { ... }
  }
};
```

Then in your templates just use `{{ page.date | formatDate(dateformats.en.short, true) }}`. At least now all your date formats can be defined globally in a single place. Or you could put them into separate files if you wanted, like src/_data/dateformats/en.js, etc.
You might even be able to use a variable for the locale in a directory data file and then maybe use:

```njk
{{ page.date | formatDate(dateformats[locale].short, true) }}
```

---

As for `{{ fundraising.contributors }}` that is a bit trickier. I'm not sure if that's an array of names or where it's coming from. If the list is the same in English and German, it might make sense to define it in a global data file (src/_data/fundraising.js which contains an array of contributor names). This way both the English and German templates would use the same names and you wouldn't need to define the data in two templates/layouts/includes.
I'd probably even try formatting the names using something like [`Intl.ListFormat`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat). In fact, the MDN docs already have an excellent example on using it w/ English and German languages.

```js
const formatter = new Intl.ListFormat('en', { style: 'long', type: 'conjunction' });
console.log(formatter.format(vehicles));
// expected output: "Motorcycle, Bus, and Car"
```

I'd probably wrap that in an Eleventy custom filter, so you could do `{{ fundraising.contributors | listformat(locale) }}` maybe.

---

I haven't done a lot of translated static sites, but it might be easier to have separate templates for whichever languages you want to support.

```njk
// src/_includes/de/fundraising_incl.njk
<p>In dieser Spendenaktion haben bereits <b class="no-line-break">{{ fundraising.contributors | listformat(locale) }} Spender</b> mitgewirkt (Stand {{ page.date | formatDate(dateformats.de.short, true) }} Uhr). Wir sagen DANKE für euren Beitrag!</p>
```

```njk
// src/_includes/en/fundraising_incl.njk
<p><b class="no-line-break">{{ fundraising.contributors | listformat(locale) }} donors</b> have already contributed to this fundraiser (as of {{ page.date | formatDate(dateformats.en.short, true) }} hour). We say THANK YOU for your contribution!</p>
```

Or if you don't want multiple include files, you could maybe just do some big `if..elif..else` block to check if you should display the English version or German version or French version, etc.
